deliver/lib/deliver/runner.rb

```diff
@@ -270,25 +270,32 @@ def submit_for_review
     # If itc_provider was explicitly specified, use it.
     # If there are multiple teams, infer the provider from the selected team name.
     # If there are fewer than two teams, don't infer the provider.
-    def transporter_for_selected_team
+    def transporter_for_selected_team(upload: false)
       # Use JWT auth
       api_token = Spaceship::ConnectAPI.token
+      api_key = if options[:api_key].nil? && !options[:api_key_path].nil?
+                  # Load api key info if user set api_key_path, not api_key
+                  { key_id: api_token.key_id, issuer_id: api_token.issuer_id, key: api_token.key_raw, is_key_content_base64: api_token.is_key_content_base64 }
+                else
+                  options[:api_key]
+                end
```

Comment:
I have no idea how the various api token / api key path options work (haven't looked into it in details), but isn't this strange here that we test on `api_key_path` being provided… but don't use its value at all here? Felt strange when I read this code and made me think there was a typo in the condition and option name at first.

Given that the `api_token = Spaceship::ConnectAPI.token` value comes from [those lines in `login` that are the ones using the `options[:api_key_path]`](https://github.com/fastlane/fastlane/blob/38f23ee3775e0fdf62c5aee013c6bc9842223ba9/deliver/lib/deliver/runner.rb#L30-L32), I'm assuming that's why it works in the end… but wouldn't it make more sense and help readability of the code to test `!api_token.nil?` instead of `!options[:api_key_path].nil?` here for consistency (since that's the value we end up using on line 278 after all)?

Reply:
> I have no idea how the various api token / api key path options work

Yes. me too. It is very hard for me to understand all 😂 

> but it feels odd here that we test on api_key_path being provided… but don't use its value at all here? (made me think there was a typo in the condition and option name at first tbh).

This represents api_token is provided if user set `api_key_path` option but `api_key_path` no longer used here. I wrote so to solve api_key_path issue which is reported (maybe, my brai focused to write to handle `api_key_path` too much)

> but wouldn't it make more sense and help readability of the code to test !api_token.nil? instead of !options[:api_key_path].nil? here for consistency (since that's the value we end up consuming on line 278 after all)?

I think it looks good idea, let me test here and commit soon

Reply:
fixed! https://github.com/fastlane/fastlane/pull/20631/commits/ab02d6dacb2c2a2b9d8ca82a5dd991a4315fbcb2